.circleci/config.yml

```diff
@@ -0,0 +1,269 @@
+  
+jobs:
+  dependencies-pr:
+    docker:
+      - <<: *dependency_bot
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          command: |
+            git config user.email "no@email.com"
+            git config user.name "Dependency bot circleci"
+            git config core.autocrlf true
+            hub --version
+      - run:
+          command: |
+            depid=$(date +%y%m%d%H%M%S)
+            git checkout -b dependency_$depid
+            bash .circleci/build_dependencies.sh
+
+            git --no-pager diff .circleci/dependencies.txt
+            set -x
+            git diff --quiet .circleci/dependencies.txt || (
+                set +o pipefail
+                curl -s https://api.github.com/repos/os-autoinst/os-autoinst/commits | grep sha | head -n 1 | grep -o -E '[0-9a-f]{40}' > .circleci/autoinst.sha
+                git add .circleci/dependencies.txt .circleci/autoinst.sha
+                git commit -m "Dependency cron $depid"
+                git push $CIRCLE_REPOSITORY_URL dependency_$depid
+                hub pull-request -m "Dependency cron $depid" --base $CIRCLE_PROJECT_USERNAME:$CIRCLE_BRANCH
+            )
+
+  build-cache:
+    docker:
+      - <<: *base
+    steps:
+      - checkout
+      - run: ls -la && pwd 
+      - run: cat .circleci/dependencies.txt
+      - run: *chown_hack_for_cache
+      - restore_cache: *restore_cache
+      - run: *show_diagnostic_info
+      - run:
+          command: |
+            if [ ! -d /var/cache/zypp/packages/openQA ]; then
+              bash -x .circleci/build_cache.sh
+            fi
+      - save_cache: *save_cache
+
+  build-integration-cache:
+    docker:
+      - <<: *base
+    steps:
+      - checkout
+      - run: *chown_hack_for_cache_integration
+      - restore_cache: *restore_cache
+      - restore_cache: *restore_integration_cache
+      - run: find /var/cache/zypp/packages/ | wc -l
+      - run:
+          command: |
+            ls -lRa /var/cache/zypp/packages/openQA || :
+            if [ ! -d /var/cache/autoinst/t ]; then
+              bash -x .circleci/build_autoinst.sh "/var/cache/autoinst"
+            fi
+      - save_cache: *save_integration_cache
+
+  test1:
+    docker:
+      - <<: *base
+    steps:
+      - checkout
+      - run: *chown_hack_for_cache
+      - restore_cache: *restore_cache
+      - run: *check_cache
+      - run: *install_cached_packages
+      - run: sudo zypper -n install python3-six
+      - run: eval "$(t/test_postgresql | grep TEST_PG=)" && PERL5LIB=lib prove -v t/*.t
+
+  testui:
+    docker:
+      - <<: *base
+    steps:
+      - checkout
+      - run: *chown_hack_for_cache
+      - restore_cache: *restore_cache
+      - run: *check_cache
+      - run: *install_cached_packages
+      - run: 
+          command: |
+            eval "$(t/test_postgresql | grep TEST_PG=)" && PERL5LIB=lib prove -v $(find t/ui/ -type f -name '*.t' | grep -vf .circleci/unstable_tests.txt | sort)
```

Comment:
Test executions are inline, rather than using a script/ Makefile like in our Travis setup. I feel like it would be nice to continue to share the commands. This would be especially important with the unstable tests if one wanted to easily re-run the unstable tests in one's local development environment.

Reply:
Right. And the "eval-grep" is also a bit hacky considering that one can simply pass a path to `t/test_postgresql` instead of letting that script decide what path to use.